packages/owl-bot/src/copy-code.ts

```diff
@@ -42,29 +42,67 @@ const readFileAsync = promisify(readFile);
 /**
  * Composes a link to the source commit that triggered the copy.
  */
-function sourceLinkFrom(sourceCommitHash: string): string {
+export function sourceLinkFrom(sourceCommitHash: string): string {
   return `https://github.com/googleapis/googleapis-gen/commit/${sourceCommitHash}`;
 }
 
 /**
- * Copies the code from googleapis-gen to the dest repo, and creates a
- * pull request.
+ * Composes the line we append to a pull request body.
+ */
+export function sourceLinkLineFrom(sourceLink: string): string {
+  return `Source-Link: ${sourceLink}\n`;
+}
+
+/**
+ * Finds the source link in a pull request body.  Returns the empty string if
+ * not found.
+ */
+export function findSourceHash(prBody: string): string {
+  const match =
+    /https:\/\/github.com\/googleapis\/googleapis-gen\/commit\/([0-9A-Fa-f]+)/.exec(
+      prBody
+    );
+  return match ? match[1] : '';
+}
+
+/**
+ * A return value indicating that code was copied into a local branch.
+ */
+interface LocalCopy {
+  kind: 'LocalCopy';
```

Comment:
Also a nit, I believe that TypeScript has `enums`, which might be worth considering for these fields.

Reply:
I thought about that too, but I don't see an advantage in this particular case.  A typo in any instance of these literal strings would trigger a compile error.